aten/src/ATen/native/Normalization.cpp

```diff
@@ -416,6 +416,22 @@ std::tuple<Tensor, Tensor, Tensor, Tensor, int64_t> _batch_norm_impl_index(
   const Tensor& running_var = c10::value_or_else(running_var_opt, [] {return Tensor();});
 
   auto num_features = input.sizes()[1];
+
+  if (input.numel() == 0) {
+    Tensor reserve = at::empty({0}, input.options().dtype(kByte));
+    auto options = input.options().dtype(
+        at::toAccumulateType(input.scalar_type(), /*is_cuda=*/input.is_cuda()));
+    auto save_mean = at::empty({num_features}, options);
```

Comment:
note than in eager mode, this creates (potentially large) Tensors that just get discarded on exit....

Reply:
Yeah, that's unfortunate.
But if we are to stick to our plan of unifying the APIs, this will happen to even wider extend... I'll open an upstream issue so we can have a discussion there!